resources/repo.rb

```diff
@@ -0,0 +1,88 @@
+property :repo_name, String, name_attribute: true
+property :repo_type, String, default: 'maven-hosted'
+property :config, Hash
+property :api_url, String
+property :api_user, String
+property :api_password, String
+
+def apiclient
+  @apiclient ||= Nexus3::Api.new(api_url, api_user, api_password)
+end
+
+def textify(args)
+  require 'json'
+  JSON.generate(args)
+end
+
+load_current_value do
+  apiclient.list_repositories
```

Comment:
The api method should be more generic like `run_script('get_repo', params)`
You should do something with the result e.g. set current_value properties